src/plots/cartesian/autorange.js

```diff
@@ -242,14 +244,34 @@ function concatExtremes(gd, ax) {
     _concat(fullLayout.annotations || [], ax._annIndices || []);
     _concat(fullLayout.shapes || [], ax._shapeIndices || []);
 
+    // Include the extremes from other matched axes with this one
+    if(ax._matchGroup && !noMatch) {
+        for(var axId2 in ax._matchGroup) {
+            if(axId2 !== ax._id) {
+                var ax2 = getFromId(gd, axId2);
+                var extremes2 = concatExtremes(gd, ax2, true);
+                // convert padding on the second axis to the first with lenRatio
+                var lenRatio = ax._length / ax2._length;
```

Comment:
Previously we calculated autorange for matched axes incorrectly: we just took the minimum and maximum of all the individually-calculated ranges. This is fine when there's no padding around the edges (pixel padding or the 5% "extrapad") but when we have either of these, the simple combination of results gives too little padding. So what I changed to instead is combining the `extremes` arrays (sets of min and max items each with its own padding) by converting pixel padding to its equivalent on one axis. Note that this is still compatible with `constrain='domain'` because, while it shrinks the axes' `_length`, it does so symmetrically across the whole matching group.